python/client/util/SessionRunner.py

```diff
@@ -0,0 +1,177 @@
+from asyncio import gather
+from cakelab.arflow_grpc.v1.session_pb2 import Session
+from cakelab.arflow_grpc.v1.device_pb2 import Device
+
+from cakelab.arflow_grpc.v1.ar_frame_pb2 import ARFrame
+from cakelab.arflow_grpc.v1.color_frame_pb2 import ColorFrame
+from cakelab.arflow_grpc.v1.xr_cpu_image_pb2 import XRCpuImage
+from cakelab.arflow_grpc.v1.vector2_int_pb2 import Vector2Int
+from google.protobuf.timestamp_pb2 import Timestamp
+import ffmpeg
+import subprocess
+import threading
+import cv2
+import time
+from typing import Callable, Coroutine, Any
+
+class SessionRunner:
+    camera : cv2.VideoCapture | None = None
+    session: Session | None = None
+    device: Device | None = None
+    onARFrame: Callable[[Session, ARFrame, Device], Coroutine[Any, Any, None]] | None = None
+    def __init__(self, session: Session, device: Device, onARFrame: Callable[[Session, ARFrame, Device], Coroutine[Any, Any, None]], gathering_interval: int):
```

Comment:
This file needs some formatting. I would recommend using VSCode + Ruff extension to format it.